# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLTextureUpload.cpp
# Commit: 51a413479f1d
# Full Hash: 51a413479f1d5d898a802e75b86a42916b769261
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2021-01-16 09:28:04
# Description:
#   Bug 1686750 - Upload zeros if layers::Image => SurfaceDescriptor fails. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D101948
# ==============================================================================

diff -r bc4560067c86 -r 51a413479f1d dom/canvas/WebGLTextureUpload.cpp
--- a/dom/canvas/WebGLTextureUpload.cpp	Fri Jan 15 19:28:12 2021 +0000
+++ b/dom/canvas/WebGLTextureUpload.cpp	Fri Jan 15 19:18:38 2021 +0000
@@ -189,6 +189,9 @@
     if (sd) {
       sd = Some(Flatten(*sd));
     }
+    if (!sd) {
+      NS_WARNING("No SurfaceDescriptor for layers::Image!");
+    }
   }
 
   RefPtr<gfx::DataSourceSurface> dataSurf;
@@ -212,7 +215,7 @@
 
   ////
 
-  if (!layersImage && !dataSurf) {
+  if (!sd && !dataSurf) {
     webgl.EnqueueWarning("Resource has no data (yet?). Uploading zeros.");
     return Some(TexUnpackBlobDesc{target, size, gfxAlphaType::NonPremult});
   }
